### PR TITLE
Add reference page for the global module

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
This PR adds a new reference documentation page for the global module. Closes PRO-2940.

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

1. Build the documentation
2. Navigate to reference/modules/global.html
3. 
## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [x] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: docs/reference/modules/global.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+extends: '@apostrophecms/piece-type'
+---
+
+# `@apostrophecms/global`
+
+**Alias:** `apos.global`
+
+<AposRefExtends :module="$frontmatter.extends" />
+
+This module populates the `req.data.global` with content or settings that are utilized site-wide. It is configured through "implicit sub-classing" of the schema fields present in the parent module. New fields are added and grouped identically to any widget or pieces module. The data from these fields is available to templates as `data.global`.
```
Comment (boutell) [inline docs/reference/modules/global.md:11]: The `req.data.global` object

Comment (boutell) [reply docs/reference/modules/global.md:11]: `@apostrophecms/global` is a piece type that only ever has one piece.

File: docs/reference/modules/global.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+If set to `true`, the `deferWidgetLoading` option reduces the number of queries required when loading the global doc. Any widget with the option `deferred: true` used on the page will not be loaded until the entire global doc is loaded.
+
+#### Example
+
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+ options: {
+   deferWidgetLoad: true
```
Comment (boutell) [inline docs/reference/modules/global.md:33]: deferWidgetLoading

File: docs/reference/modules/global.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+       label: 'Organization GitHub url'
+     },
+     footerMissionStatement: {
+       type: 'area',
+       label: 'Footer mission statement',
+       options: {
+         widgets: {
+           '@apostrophecms/rich-text': {}
+         }
+         // Other fields with multiple widget types
```
Comment (boutell) [inline docs/reference/modules/global.md:48]: This comment is confusing. I think it's clear you can add more than one field, more than one widget type, etc. to someone who's already been through the basics of the guide, which should be a safe assumption in the reference docs.

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/reference/modules/global.md (hunk 1 above)
Comment (BoDonkey) [reply docs/reference/modules/global.md:11]: Changed for the first and third comments. For the second comment, I'm assuming you are referring to the note?

File: docs/reference/modules/global.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+# `@apostrophecms/global`
+
+**Alias:** `apos.global`
+
+<AposRefExtends :module="$frontmatter.extends" />
+
+This module populates the `req.data.global` with content or settings that are utilized site-wide. It is configured through "implicit sub-classing" of the schema fields present in the parent module. New fields are added and grouped identically to any widget or pieces module. The data from these fields is available to templates as `data.global`.
+
+::: note
+There can only be *one* global doc.
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/reference/modules/global.md:14]: Changed based on the first comment block.

File: docs/reference/modules/global.md (hunk 2 above)
Comment (BoDonkey) [reply docs/reference/modules/global.md:33]: Changed.

File: docs/reference/modules/global.md (hunk 3 above)
Comment (BoDonkey) [reply docs/reference/modules/global.md:48]: Extra comment eliminated.

**Review by boutell — APPROVED** [review]

**Pull request merged**